File: src/main/kotlin/wowsft/model/gameparams/ship/component/torpedo/TorpedoAmmo.java
package wowsft.model.gameparams.ship.component.torpedo;

import wowsft.config.WoWSFT;
import wowsft.model.gameparams.TypeInfo;
import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.List;

@Data
@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
public class TorpedoAmmo
{
    private List<Object> additionalShips;
    private float alertDist;
    private float alphaDamage;
    private float alphaPiercingHE;
    private String ammoType;
    private float armingTime;
    private float bulletDiametr;
    private float burnProb;
    private float costCR;
    private float damage;
    private float damageUW;
    private float depth;
    private float directDamage;
    private float fallingTimeCoef;
    private float id;
    private List<String> ignoreClasses;
    private String index;
    @JsonAlias("isDeepWater")
    private boolean deepWater;
    private float maxDist;
    private String name;
    private String planeAmmoType;
    private float speed;
    private float splashArmorCoeff;
    private float splashCubeSize;
    private TypeInfo typeinfo;
    private boolean uwAbility;
    private float uwCritical;
    private float visibilityFactor;
    private float volume;
}

File: src/main/kotlin/wowsft/model/gameparams/ship/component/torpedo/TorpedoAmmo.kt
package wowsft.model.gameparams.ship.component.torpedo

import com.fasterxml.jackson.annotation.JsonAlias
import wowsft.config.WoWSFT
import wowsft.model.gameparams.TypeInfo
import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class TorpedoAmmo {
    var additionalShips = ArrayList<Any>()
    var alertDist = 0.0
    var alphaDamage = 0.0
    var alphaPiercingHE = 0.0
    var ammoType: String? = null
    var armingTime = 0.0
    var bulletDiametr = 0.0
    var burnProb = 0.0
    var costCR = 0.0
    var damage = 0.0
    var damageUW = 0.0
    var depth = 0.0
    var directDamage = 0.0
    var fallingTimeCoef = 0.0
    var id: Long = 0
    var ignoreClasses = ArrayList<String>()
    var index: String? = null
    @JsonAlias("isDeepWater")
    var deepWater = false
    var maxDist = 0.0
    var name: String? = null
    var planeAmmoType: String? = null
    var speed = 0.0
    var splashArmorCoeff = 0.0
    var splashCubeSize = 0.0
    var typeinfo : TypeInfo? = null
    var uwAbility = false
    var uwCritical = 0.0
    var visibilityFactor = 0.0
    var volume = 0.0
}
